iclaims.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021-2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package ccatoken
+
+import (
+	"fmt"
+)
+
+// IClaims provides a uniform interface for dealing with CCA realm claims
+type IClaims interface {
+	// Getters
+	GetChallenge() ([]byte, error)
+	GetPersonalizationValue() ([]byte, error)
+	GetInitialMeasurements() ([]byte, error)
+	GetExtensibleMeasurements() ([][]byte, error)
+	GetHashAlgID() (string, error)
+	GetPubKey() ([]byte, error)
+	GetPubKeyHashAlgID() (string, error)
+
+	// Setters
+	SetChallenge([]byte) error
+	SetPersonalizationValue([]byte) error
+	SetInitialMeasurements([]byte) error
+	SetExtensibleMeasurements([][]byte) error
+	SetHashAlgID(string) error
+	SetPubKey([]byte) error
+	SetPubKeyHashAlgID(string) error
+
+	// CBOR codecs
+	FromCBOR([]byte) error
+	ToCBOR() ([]byte, error)
+
+	// JSON codecs
+	FromJSON([]byte) error
+	ToJSON() ([]byte, error)
+
+	// Semantic validation
+	Validate() error
+}
+
+func NewClaims() (IClaims, error) {
+	return &CcaRealmClaims{}, nil
+}
+
+func DecodeClaims(buf []byte) (IClaims, error) {
+	cl := &CcaRealmClaims{}
+
+	if err := cl.FromCBOR(buf); err != nil {
+		return nil, err
+	}
+
+	return cl, nil
+}
+
+func DecodeJSONClaims(buf []byte) (IClaims, error) {
+	cl := &CcaRealmClaims{}
+
+	if err := cl.FromJSON(buf); err != nil {
+		return nil, err
+	}
+
+	return cl, nil
+}
+
+func validate(c IClaims) error {
+
+	// realm challenge
+	_, err := c.GetChallenge()
+	if err != nil {
+		return fmt.Errorf("validating cca-realm-challege claim : %w", err)
```

Comment:
spurious whitespace (here and below)

```suggestion
		return fmt.Errorf("validating cca-realm-challege claim: %w", err)
```